editor/src/core/model/element-template-utils.ts

```diff
@@ -543,6 +544,92 @@ export function insertJSXElementChild(
   return insertChildAndDetails(updatedComponents, null, importsToAdd) // TODO is this wrapper type needed?
 }
 
+export function insertJSXElementChildren(
+  projectContents: ProjectContentTreeRoot,
+  targetParent: InsertionPath,
+  elementsToInsert: Array<JSXElementChild>,
+  components: Array<UtopiaJSXComponent>,
+  indexPosition: IndexPosition | null,
+): InsertChildAndDetails {
+  const parentPath: StaticElementPath = targetParent.intendedParentPath
+  let importsToAdd: Imports = {}
+  const updatedComponents = transformJSXComponentAtPath(components, parentPath, (parentElement) => {
+    if (isChildInsertionPath(targetParent)) {
+      if (!isJSXElementLike(parentElement)) {
+        throw new Error("Target parent for child element insertion doesn't support children")
+      }
+      let updatedChildren: Array<JSXElementChild>
+      if (indexPosition == null) {
+        updatedChildren = [...parentElement.children, ...elementsToInsert]
+      } else {
+        updatedChildren = elementsToInsert.reduce((working, elementToInsert) => {
+          return addToArrayAtIndexPosition(elementToInsert, working, indexPosition)
+        }, parentElement.children)
+      }
+      return {
+        ...parentElement,
+        children: updatedChildren,
+      }
+    } else if (isConditionalClauseInsertionPath(targetParent)) {
+      if (!isJSXConditionalExpression(parentElement)) {
+        throw new Error('Target parent for conditional insertion is not conditional expression')
+      }
+      // Determine which clause of the conditional we want to modify.
+      const conditionalCase = targetParent.clause
+      const toClauseOptic =
+        conditionalCase === 'true-case' ? conditionalWhenTrueOptic : conditionalWhenFalseOptic
+
+      function wrapIntoFragment(clauseValue: JSXElementChild | null): JSXFragment {
```

Comment:
I think the null could be removed here if this function was passed the new children array (so clauseValue + the array to be inserted if there's an existing clause value, or just the array to be inserted if the clause value is null)